### PR TITLE
Add `core.contiguousLength`

### PR DESCRIPTION
**Pull request by kasperisager**
This PR adds a `core.contiguousLength` property, stored in the oplog header, that tracks the number of contiguously available blocks starting from the first block of a core.

### CODE REVIEW

File: test/helpers/index.js [hunk 1]
```diff
@@ -1,38 +1,36 @@
...
+  s1.pipe(s2).pipe(s1)
+  return [s1, s2]
+}
 
-  async eventFlush () {
-    await new Promise(resolve => setImmediate(resolve))
-  }
+exports.eventFlush = async function eventFlush () {
+  await new Promise(resolve => setImmediate(resolve))
 }
```
Comment (kasperisager) [inline test/helpers/index.js:36]: This ensures that the functions are available as synthesised ESM exports, making sandbox scripts are little easier to write.

**Comment by mafintosh** [comment]
Excellent! I think there is an edge case when the feed is truncated, can you test that?

We need an additional feature to this. A new flag `sparse` that defaults to `true`, that when `false` makes .length === .contigiousLength and also sets byteLength to the equi. We can look into that in another PR tho.

File: test/helpers/index.js (hunk 1 above)
Comment (mafintosh) [reply test/helpers/index.js:36]: 👍

File: lib/core.js [hunk 2]
```diff
@@ -469,6 +469,7 @@ module.exports = class Core {
 
     const appended = batch.length > batch.ancestors
 
+    this.header.contiguousLength = batch.length
```
Comment (mafintosh) [inline lib/core.js:472]: I think this should be Math.min(batch.ancestors, contigLength)

Comment (mafintosh) [reply lib/core.js:472]: You can test that by checking the contig len on a peer that is replicating if the writable one truncates
